pkg/query-service/rules/thresholdRule.go

```diff
@@ -764,6 +766,23 @@ func (r *ThresholdRule) buildAndRunQuery(ctx context.Context, ts time.Time, ch c
 	return nil, fmt.Errorf("this is unexpected, invalid query label")
 }
 
+func normalizeLabelName(name string) string {
+	// See https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+
+	// Regular expression to match non-alphanumeric characters except underscores
+	reg := regexp.MustCompile(`[^a-zA-Z0-9_]`)
```

Comment:
Here we are trying to replace everything that's not following the spec and the following line checks if the first letter is not alphabet it will be prefixed with an nderscore.

Reply:
The return value of this function will match the regex `[a-zA-Z_][a-zA-Z0-9_]* `. Happy to update the code If there is a simpler way to achieve the same.

Reply:
it's simple the current way